src/components/canvas-tools/image-tool.tsx

```diff
@@ -85,24 +107,18 @@ export default class ImageToolComponent extends BaseComponent<IProps, {}> {
     );
   }
 
-  private handleImageUrlError = () => {
-
-    const { hasUpdatedUrl } = this.state;
-    const { db } = this.stores;
-    const { model } = this.props;
-    const { content } = model;
-    if (!hasUpdatedUrl) {
-      const imageContent = content as ImageContentModelType;
-      let updatedUrl = imageContent.url;
-      fetchImageUrl(imageContent.url, db.firebase, (fullUrl: string) => {
-        updatedUrl = fullUrl;
+  private handleUpdateImageDimensions = (imageDisplayUrl: string) => {
+    this._asyncRequest = getDimensions(undefined, imageDisplayUrl).then((dimensions: any) => {
+      this._asyncRequest = null;
```

Comment:
```suggestion
      // in case we were unmounted
      if (this._asyncRequest) {
        this._asyncRequest = null;
        this.setState({
          imageUrl: imageDisplayUrl,
          imageDimensions: dimensions,
          isLoading: false
        });
      }
```